### ISSUE TITLE
`Style/For` cannot correct `for` loop without `do`/`;`

### ISSUE THREAD

**Issue by autopp** [rubocop-hq/rubocop#6296]
`Style/For` Cop cannot correct automatically when `for` loop dose not have `do`.

## Expected behavior

When `EnforcedStyle` is `each`,

```ruby
for x in ary
  p x
end
```

is corrected to:

```ruby
ary.each do |x|
  p x
end
```

## Actual behavior

A error is occured
```
$ bundle exec rubocop --auto-correct -d sample.rb
For /Users/autopp/ruby/rubocop: configuration from /Users/autopp/ruby/rubocop/.rubocop.yml
configuration from /Users/autopp/ruby/rubocop/vendor/bundle/ruby/2.5.0/gems/rubocop-rspec-1.29.1/config/default.yml
configuration from /Users/autopp/ruby/rubocop/vendor/bundle/ruby/2.5.0/gems/rubocop-rspec-1.29.1/config/default.yml
Default configuration from /Users/autopp/ruby/rubocop/config/default.yml
Inheriting configuration from /Users/autopp/ruby/rubocop/config/enabled.yml
Inheriting configuration from /Users/autopp/ruby/rubocop/config/disabled.yml
Inheriting configuration from /Users/autopp/ruby/rubocop/.rubocop_todo.yml
Inspecting 1 file
Scanning /Users/autopp/ruby/rubocop/sample.rb
An error occurred while Style/For cop was inspecting /Users/autopp/ruby/rubocop/sample.rb:3:0.
undefined method `end_pos' for nil:NilClass
/Users/autopp/ruby/rubocop/lib/rubocop/cop/style/for.rb:102:in `autocorrect_to_each'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/style/for.rb:86:in `autocorrect'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/cop.rb:153:in `correct'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/cop.rb:131:in `add_offense'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/style/for.rb:63:in `on_for'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/commissioner.rb:58:in `block (2 levels) in trigger_responding_cops'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/commissioner.rb:106:in `with_cop_error_handling'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/commissioner.rb:57:in `block in trigger_responding_cops'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/commissioner.rb:56:in `each'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/commissioner.rb:56:in `trigger_responding_cops'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/commissioner.rb:34:in `block (2 levels) in <class:Commissioner>'
/Users/autopp/ruby/rubocop/lib/rubocop/ast/traversal.rb:13:in `walk'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/commissioner.rb:46:in `investigate'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/team.rb:115:in `investigate'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/team.rb:95:in `offenses'
/Users/autopp/ruby/rubocop/lib/rubocop/cop/team.rb:44:in `inspect_file'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:269:in `inspect_file'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:216:in `block in do_inspection_loop'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:248:in `block in iterate_until_no_changes'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:241:in `loop'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:241:in `iterate_until_no_changes'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:212:in `do_inspection_loop'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:115:in `block in file_offenses'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:133:in `file_offense_cache'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:113:in `file_offenses'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:101:in `process_file'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:78:in `block in each_inspected_file'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:75:in `each'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:75:in `reduce'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:75:in `each_inspected_file'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:67:in `inspect_files'
/Users/autopp/ruby/rubocop/lib/rubocop/runner.rb:39:in `run'
/Users/autopp/ruby/rubocop/lib/rubocop/cli.rb:157:in `execute_runner'
/Users/autopp/ruby/rubocop/lib/rubocop/cli.rb:85:in `execute_runners'
/Users/autopp/ruby/rubocop/lib/rubocop/cli.rb:41:in `run'
/Users/autopp/ruby/rubocop/exe/rubocop:13:in `block in <top (required)>'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/Users/autopp/ruby/rubocop/exe/rubocop:12:in `<top (required)>'
/Users/autopp/ruby/rubocop/vendor/bundle/ruby/2.5.0/bin/rubocop:23:in `load'
/Users/autopp/ruby/rubocop/vendor/bundle/ruby/2.5.0/bin/rubocop:23:in `<top (required)>'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `load'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `kernel_load'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:28:in `run'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:424:in `exec'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:27:in `dispatch'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:18:in `start'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/exe/bundle:30:in `block in <top (required)>'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/friendly_errors.rb:122:in `with_friendly_errors'
/Users/autopp/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/bundler-1.16.1/exe/bundle:22:in `<top (required)>'
/Users/autopp/.rbenv/versions/2.5.1/bin/bundle:23:in `load'
/Users/autopp/.rbenv/versions/2.5.1/bin/bundle:23:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/For cop was inspecting /Users/autopp/ruby/rubocop/sample.rb:3:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop-hq/rubocop/issues

Mention the following information in the issue report:
0.59.0 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-darwin16)
Finished in 0.26160600000002887 seconds
```

## Steps to reproduce the problem

Set `EnforcedStyle` of `Style/For` Cop to `each` and try correct `for` loop without `do`.

```
$ cat sample.rb
for x in ary
  p x
end

$ bundle exec rubocop -d --only Style/For --auto-correct sample.rb
```

## RuboCop version

```
$ bundle exec rubocop -V
0.59.0 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-darwin16)
```
